lib/reversetunnel/localsite.go

```diff
@@ -497,10 +497,120 @@ func (s *localSite) periodicFunctions() {
 			if err := s.sshTunnelStats(); err != nil {
 				s.log.Warningf("Failed to report SSH tunnel statistics for: %v: %v.", s.domainName, err)
 			}
+			s.agentStats()
 		}
 	}
 }
 
+// agentStats leverages the accessPoint cache to report all agents/proxies/auth servers connected to the
+// teleport cluster via prometheus metrics
+func (s *localSite) agentStats() {
+	hostID := make(map[string]struct{})
+	versionCount := make(map[string]int)
+
+	// Nodes, Proxies, Auths, KubeServices, and WindowsDesktopServers use the UUID as the name field where
+	// DB and App store it in the spec. Check expiry due to DynamoDB taking up to 48hr to expire from backend
+	// and then store hostID and version count information.
+	serverCheck := func(server interface{}) {
+		type serverKubeWindows interface {
+			Expiry() time.Time
+			GetName() string
+			GetTeleportVersion() string
+		}
+		type appDB interface {
+			serverKubeWindows
+			GetHostID() string
+		}
+
+		ttl := s.clock.Now().Add(-1 * apidefaults.ServerAnnounceTTL)
```

Comment:
nit: I would expect something called `ttl` to be of type `time.Duration`. This is a `time.Time`, so maybe something like `expiration` would be a better name.

Reply:
Misunderstood what this was doing in `sshTunnelStats()` and now questioning why it is there. I have removed this in lieu of just checking against the current time if the resource has expired.